grails-app/services/org/pih/warehouse/data/ProductSynonymDataService.groovy

```diff
@@ -65,12 +68,16 @@ class ProductSynonymDataService {
             // Check if none error occures to be sure, that we have correct product, locale and synonymTypeCode
             // before we check the duplicates
             if (!command.errors.allErrors && synonymTypeCode == SynonymTypeCode.DISPLAY_NAME) {
-               Set<Synonym> duplicates = product?.synonyms?.findAll { Synonym synonym ->
-                   synonym.locale == new Locale(params['locale']) && synonym.synonymTypeCode == SynonymTypeCode.DISPLAY_NAME
-               }
-               if (duplicates) {
-                   command.errors.reject("Row ${index + 1}: You cannot add multiple display names in the same language. Edit the existing synonym instead.")
-               }
+                Set<Synonym> duplicates = product?.synonyms?.findAll { Synonym synonym ->
```

Comment:
This is probably a non-problem, but I'll just mention it.
We probably don't need to do `synonyms.findAll{}` to check for duplicate DISPLAY_NAME synonym. 
If it returns at least one item we can be sure that this product already has a displayName synonym in this locale, so I guess doing this should be enought
```
Synonym displayNameSynonym = product?.synonyms?.find { Synonym synonym ->
    synonym.locale == new Locale(params['locale']) && synonym.synonymTypeCode == SynonymTypeCode.DISPLAY_NAME
}
if (displayNameSynonym || productsWithDisplayName.find{ it?.code == product?.productCode && it?.locale == params['locale'] }) { 
   ... 
}
```

Reply:
you are right, thanks for catching that